modules/core/arrow-typeclasses/src/main/kotlin/arrow/typeclasses/MonadContinuations.kt

```diff
@@ -24,8 +24,12 @@ open class MonadContinuation<F, A>(M: Monad<F>, override val context: CoroutineC
     returnedMonad = value
   }
 
+  @Suppress("UNCHECKED_CAST")
   override fun resumeWithException(exception: Throwable) {
```

Comment:
Is this respected by inheritors?

Reply:
`MonadThrow` overwrites it to capture the `Throwable` instead of throwing it.

```
  @Suppress("UNCHECKED_CAST")
  override fun resumeWithException(exception: Throwable) {
    returnedMonad = when (exception) {
      is ContinuationShortcircuitThrowable -> exception.exit as Kind<F, A>
      else -> raiseError(exception)
    }
  }
```